framework/auth/core.py

```diff
@@ -498,6 +498,9 @@ def create_unregistered(cls, fullname, email=None):
 
     @classmethod
     def create(cls, username, password, fullname):
+        if username.split('@')[1] in settings.BLACKLISTED_DOMAINS:
```

Comment:
Any reason not to just implement the `validate_email` helper from `framework/auth/utils.py`?